File: Chef and Happiness.cpp
#include<iostream>
#include<vector>
#include<set>

using namespace std;

vector<int> cnt;
vector<vector<int> > pos;
set<int> s;

void init(int n){
    cnt.clear();
    pos.clear();
    s.clear();
    cnt.resize(n+1);
    pos.resize(n+1);
    for(int &i: cnt) i = 0;
}

int main(){
    int t, n, temp;
    cin >> t;
    while(t--){
        cin >> n;
        init(n);
        for(int i = 1; i <= n; i++){
            cin >> temp;
            cnt[temp]++;
            pos[temp].push_back(i);
            s.insert(temp);
        }
        bool found = false;
        for(int i: s){
            if(cnt[i] > 1){
                int need = 2;
                for(int p: pos[i]){
                    if(cnt[p] > 0) need--;
                    if(need <= 0){
                        found = true;
                        break;
                    }
                }
                if(found) break;
            }
        }
        if(found) cout << "Truly Happy" << endl;
        else cout << "Poor Chef" << endl;
    }
}

File: Chef and Happiness.py
for _ in range(int(input())):
    n = int(input())
    a = [int(x)-1 for x in input().split()]
    pos = []
    cnt = []
    for _ in range(n):
        pos.append([])
        cnt.append(0)
    for i in range(n):
        pos[a[i]].append(i)
        cnt[a[i]]+=1
    found = False
    s = set(a)
    for i in s:
        if cnt[i] > 1:
            needed = 2
            for p in pos[i]:
                if cnt[p] > 0:
                    needed-=1
                if needed <= 0:
                    found = True
                    break
            if found: break
    
    if found: print('Truly Happy')
    else: print("Poor Chef")
